### ISSUE TITLE
CloudFormation - Unable to use intrinsic functions in output export name

### ISSUE THREAD

**Issue by Tobolov** [localstack/localstack#3301]
# Type of request: This is a ...

[ X] bug report
[ ] feature request

# Detailed description
Intrinsic functions (e.g. Sub, Join) do not work within the export name of an output.

## Expected behaviour
Stack's output in describe-stacks api response:
```json
{
    "OutputKey": "VpcId",
    "OutputValue": "<irrelevant>",
    "ExportName": "ci-vpc-id"
}
```

## Actual behaviour
Stack's output in describe-stacks api response:
```json
{
    "OutputKey": "VpcId",
    "OutputValue": "<irrelevant>",
    "ExportName": "{'Fn::Sub': '${Environment}-vpc-id'}"
}
```

# Steps to reproduce
1. Deploy the following CloudFormation template:
```yaml
---
AWSTemplateFormatVersion: '2010-09-09'
Parameters:
  Environment:
    Type: String
    Default: 'ci'
Resources:
  VPC:
    Type: AWS::EC2::VPC
    Properties:
      EnableDnsSupport: true
      EnableDnsHostnames: true
      CidrBlock: "100.0.0/20"
Outputs:
  VpcId:
    Value: !Ref VPC
    Export:
      Name: !Sub '${Environment}-vpc-id'

```
2. Run aws cloudformation describe-stacks --endpoint-url <localstack-endpoint>

**Comment by Tobolov** [comment]
Another example of this bug:
```yaml
---
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  InstanceRole:
    Properties:
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - ec2.amazonaws.com
                - ssm.amazonaws.com
            Action:
              - sts:AssumeRole
      Policies: 
        - PolicyName: "RegmonInstancePolicy"
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: Allow
                Resource: '*'
                Action:
                - ec2:GetPasswordData
Outputs:
  RoleArn:
    Value: !GetAtt InstanceRole.Arn
    Export:
      Name: RoleArn
```

Running `aws cloudformation describe-stacks --endpoint-url <url>` works and the stack output is correct.
However, the api response from creating the cloudformation template is 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<ErrorResponse xmlns="http://cloudformation.amazonaws.com/doc/2010-05-15/">
   <Error>
      <Type>Sender</Type>
      <Code>AlreadyExistsException</Code>
      <Message>Stack [temp] already exists</Message>
   </Error>
   <RequestId>950ff8d7-812a-44b3-bb0c-9b271b954104</RequestId>
</ErrorResponse>
```

The error found in localstack's logs is as follows:
```
Traceback (most recent call last):
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 323, in run_wsgi
    execute(self.server.app)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 312, in execute
    application_iter = app(environ, start_response)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/server.py", line 173, in __call__
    return backend_app(environ, start_response)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/core/utils.py", line 151, in __call__
    result = self.callback(request, request.url, {})
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/core/responses.py", line 202, in dispatch
    return cls()._dispatch(*args, **kwargs)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/core/responses.py", line 312, in _dispatch
    return self.call_action()
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/core/responses.py", line 397, in call_action
    response = method()
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/responses.py", line 95, in create_stack
    stack = self.cloudformation_backend.create_stack(
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/models.py", line 589, in create_stack
    self._validate_export_uniqueness(new_stack)
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/models.py", line 761, in _validate_export_uniqueness
    new_stack_export_names = [x.name for x in stack.exports]
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/models.py", line 346, in exports
    return self.output_map.exports
  File "/home/tobolov/linrepos/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 774, in exports
    cleaned_value = clean_json(value.get("Value"), self._resource_map)
  File "/home/tobolov/linrepos/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 306, in clean_json
    raise DependencyNotYetSatisfied(
localstack.services.cloudformation.cloudformation_starter.DependencyNotYetSatisfied: Unable to resolve attribute ref ['InstanceRole', 'Arn']
```

**Comment by whummer** [comment]
Thanks for reporting @Tobolov . This should be fixed in #3355 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by Tobolov** [comment]
Tested and working.
Thanks!
